source/init/mus_initFluid_module.f90

```diff
@@ -277,17 +217,136 @@ subroutine mus_init_advRel_fluid( relaxation, layout, compute )
+    procedure( kernel ), pointer, intent(out) :: compute
+    ! --------------------------------------------------------------------------
+    select case(trim(variant))
+    case ('standard')
+      select case (trim(layout))
+      case ('d3q27')
+        compute => mus_advRel_kCFD_rBGK_vStd_lD3Q27
+      case ('d3q19')
+        compute => mus_advRel_kFluid_rBGK_vStd_lD3Q19
+      case ('d2q9')
+        compute => mus_advRel_kFluid_rBGK_vStd_lD2Q9
+      case default
+        call tem_abort('Unsupported variant "'//trim(variant)//'" for'// &
+          &             ' "bgk" relaxation!')
+      end select
+
+    case ('standard_no_opt')
+      compute => mus_advRel_kCFD_rBGK_vStdNoOpt_l
+
+    case ('improved')
+      select case (trim(layout))
+      case ('d3q27')
+        compute => mus_advRel_kFluid_rBGK_vImproved_lD3Q27
+      case ('d2q9')
+        compute => mus_advRel_kFluid_rBGK_vImproved_lD2Q9
+      case default
+        call tem_abort('Stencil '//trim(layout)//' is not supported yet!')
+      end select
+
+    case ('block')
+      select case (trim(layout))
+      case ('d3q19')
+        compute => mus_advRel_kFluid_rBGK_vBlock_lD3Q19
+      case default
+        call tem_abort('Stencil '//trim(layout)//' is not supported yet!')
+      end select
+
+    case default
+      call tem_abort('Unsupported variant "'//trim(variant)//'" for'// &
+        &             ' "bgk" relaxation!')
+    end select
+
+  end subroutine mus_init_advRel_fluid_bgk
+  ! ************************************************************************** !
+
+  ! ************************************************************************** !
+  !> This routine assigns compute routine for mrt relaxation
+  subroutine mus_init_advRel_fluid_mrt( variant, layout, compute )
+    ! --------------------------------------------------------------------------
+    character(len=labelLen), intent(in) :: variant
+    character(len=labelLen), intent(in) :: layout
+    procedure( kernel ), pointer, intent(out) :: compute
+    ! --------------------------------------------------------------------------
+    select case (trim(variant))
+    case ('standard')
+      select case (trim(layout))
+      case ('d3q27')
+        compute => mus_advRel_kFluid_rMRT_vStd_lD3Q27
+      case ('d3q19')
+        compute => mus_advRel_kFluid_rMRT_vStd_lD3Q19
+      case ('d2q9')
+        compute => mus_advRel_kFluid_rMRT_vStd_lD2Q9
+      case default
+        call tem_abort('Unsupported variant "'//trim(variant)//'" for'// &
+          &            ' "mrt" relaxation!')
+      end select
+
+    case ('standard_no_opt')
+      select case (trim(layout))
+      case ('d3q27')
+        compute => mus_advRel_kCFD_rMRT_vStdNoOpt_lD3Q27
+      case ('d3q19')
+        compute => mus_advRel_kFluid_rMRT_vStdNoOpt_lD3Q19
+      case default
+        compute => mus_advRel_kCFD_rMRT_vStdNoOpt_l
+      end select
+
+
+    case ('bgk')
+      compute => mus_advRel_kCFD_rMRT_vStdNoOpt_l
```

Comment:
Is this an unintended leftover, or do we have a `bgk` variant of `MRT`? It seems to point to the `standard_no_opt` default of MRT, which is kind of confusing to me. Please remove if it is a just a leftover.

Reply:
It is not a leftover. It is an intended behaviour to have `bgk `as variant of `mrt`. We use the `standard_no_opt` compute kernel because it a generic one which supports this option. In `mus_mrtRelaxation_module`, all mrt relaxation parameters are set to same to recover bgk variant of mrt.